lib/msf/core/handler.rb

```diff
@@ -245,11 +245,7 @@ def register_session(session)
 
     # Call the handler's on_session() method
     on_session(session)
-
-    # Process the auto-run scripts for this session
-    if session.respond_to?('process_autoruns')
-      session.process_autoruns(datastore)
-    end
+    session.bootstrap(datastore)
```

Comment:
A problem with this, and master, is that they both call the on_session handlers before the session is properly setup. This means that crypttlv negotiation happens _after_ things like file_dropper have their way with the session, and things get all out of order. This isn't noticeable behavior-wise for most Meterpreters, but Mettle is coded to expect a crypttlv packet as the first one, or it will hang the connection.

On master currently, this patch fixes it:
```
--- a/lib/msf/base/sessions/meterpreter_options.rb
+++ b/lib/msf/base/sessions/meterpreter_options.rb
@@ -31,8 +31,6 @@ module MeterpreterOptions
   # advanced option is set to true.
   #
   def on_session(session)
-    super
-
     # Defer the session initialization to the Session Manager scheduler
     framework.sessions.schedule Proc.new {

@@ -91,14 +89,15 @@ module MeterpreterOptions
       end
     end

-    # Terminate the session without cleanup if it did not validate
-    if not valid
+    if !valid
+      # Terminate the session without cleanup if it did not validate
       session.skip_cleanup = true
       session.kill
+    else
+      # Call registered on_session callbacks
+      super
     end
-
     }
-
   end

 end
```
but, a different approach is needed with this reorganization.